autobahn/wamp_tcp_client.hpp

```diff
@@ -0,0 +1,111 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2014 Tavendo GmbH
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef AUTOBAHN_TCP_CLIENT_H
+#define AUTOBAHN_TCP_CLIENT_H
+
+#include <autobahn/autobahn.hpp>
+#include <boost/asio.hpp>
+#include <boost/version.hpp>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <tuple>
+namespace autobahn {
+/**
+ * @brief a class that wraps initialization of a raw tcp socket and WAMP client session utilizing it
+ */
+    class wamp_tcp_client {
+        typedef autobahn::wamp_session<boost::asio::ip::tcp::socket, boost::asio::ip::tcp::socket> wamp_tcp_session_t;
+    public:
+        wamp_tcp_client(std::shared_ptr<boost::asio::io_service> pIo,
+                        const boost::asio::ip::tcp::endpoint &Endpoint,
+                        const std::string &Realm,
+                        bool Debug = false) :
+                m_realm(Realm),
+                m_rawsocket_endpoint(Endpoint),
+                m_pIo(pIo),
+                m_debug(Debug) {
+            init();
```

Comment:
You could get rid of quite a few member variables (which are duplicated in wamp_session anyway) if you didn't have to call `init()`. By using [inheriting constructors](https://en.wikipedia.org/wiki/C%2B%2B11#Object_construction_improvement), you can make one constructor forward to the other and the `init()` is unnecessary, and no member variables need to exist in this class.

That said, not even asio::tcp::socket itself provides a convenience constructor for string/uint endpoints. I'll leave it up to Dave to decide whether or not this class should even have this here.